src/plot/highlight.js

```diff
@@ -0,0 +1,270 @@
+    brushArea: {top: 0, right: 0, bottom: 0, left: 0},
+    brushing: false,
+    startLocX: 0,
+    startLocY: 0,
+    dragArea: null
+  };
+
+  _getDrawArea(xLoc, yLoc) {
+    const {startLocX, startLocY} = this.state;
+    const {allow, innerWidth, innerHeight, marginLeft, marginRight, marginBottom, marginTop} = this.props;
+    const plotHeight = innerHeight + marginTop + marginBottom;
+    const plotWidth = innerWidth + marginLeft + marginRight;
+
+    return {
+      bottom: allow.includes('y') ? Math.max(startLocY, yLoc) : plotHeight,
+      left: allow.includes('x') ? Math.min(xLoc, startLocX) : 0,
+      right: allow.includes('x') ? Math.max(startLocX, xLoc) : plotWidth,
+      top: allow.includes('y') ? Math.min(yLoc, startLocY) : 0
+    };
+  }
+
+  _getDragArea(xLoc, yLoc) {
+    const {allow} = this.props;
+    const {startLocX, startLocY, dragArea} = this.state;
+
+    return {
+      bottom: dragArea.bottom + (allow.includes('y') ? (yLoc - startLocY) : 0),
+      left: dragArea.left + (allow.includes('x') ? (xLoc - startLocX) : 0),
+      right: dragArea.right + (allow.includes('x') ? (xLoc - startLocX) : 0),
+      top: dragArea.top + (allow.includes('y') ? (yLoc - startLocY) : 0)
+    };
+  }
+
+  _clickedOutsideDrag(xLoc, yLoc) {
+    const {dragArea, brushArea} = this.state;
+    const clickedOutsideDragX = dragArea && ((xLoc < brushArea.left) || (xLoc > brushArea.right));
+    const clickedOutsideDragY = dragArea && ((yLoc < brushArea.top) || (yLoc > brushArea.bottom));
+    return clickedOutsideDragX || clickedOutsideDragY;
+  }
+
+  _convertAreaToCoordinates(brushArea) {
+    const {marginLeft, allow} = this.props;
+    const xScale = getAttributeScale(this.props, 'x');
+    const yScale = getAttributeScale(this.props, 'y');
+
+    const allowX = allow.includes('x');
+    const allowY = allow.includes('y');
+
+    // Ensure that users wishes are being respected about which scales are evaluated
+    // this is specifically enabled to ensure brushing on mixed categorical and linear
+    // charts will run as expected
+
+    if (!allowX && !allowY) {
+      return {};
+    }
+
+    if (!allowX) {
+      return {
+        bottom: yScale.invert(brushArea.bottom),
+        top: yScale.invert(brushArea.top)
+      };
+    }
+
+    if (!allowY) {
+      return {
+        left: xScale.invert(brushArea.left - marginLeft),
+        right: xScale.invert(brushArea.right - marginLeft)
+      };
+    }
+
+    // NOTE only continuous scales are supported for brushing/getting coordinates back
+    return {
+      bottom: yScale.invert(brushArea.bottom),
+      left: xScale.invert(brushArea.left - marginLeft),
+      right: xScale.invert(brushArea.right - marginLeft),
+      top: yScale.invert(brushArea.top)
+    };
+  }
+
+  onParentMouseDown(e) {
+    const {onBrushStart, onDragStart, drag} = this.props;
+    const {dragArea} = this.state;
+    const {xLoc, yLoc} = getLocs(e.nativeEvent);
+
+    const startArea = (dragging, resetDrag) => {
```

Comment:
d3 calls this brushing. Why are we introducing a new variable name and concept of dragging?

Reply:
Huh. I had thought of the action of creating a selection rectangle as the brushing step and the moving the box around as a dragging. These feel like independent behaviors to me, and browsing the d3 docs it appears that they are kinda lumped under a single term. Would an appropriate rotation of terms see

brushing -> drawing
dragging -> brushing

?

Or are you suggesting a larger change?